source/shadow.c

```diff
@@ -320,14 +456,22 @@ static ShadowStatus_t extractShadowRootAndName( const char * pTopic,
                                      topicLength - *pConsumedTopicLength,
                                      pShadowNameLength );
 
-        if( shadowStatus == SHADOW_SUCCESS )
+        if( shadowStatus != SHADOW_SUCCESS )
         {
-            *pConsumedTopicLength += *pShadowNameLength;
+            shadowStatus = SHADOW_SHADOWNAME_PARSE_FAILED;
+            LogDebug( ( "Not related to Shadow. Failed to parse shadow name in pTopic %s.", pTopic ) );
```

Comment:
See previous comment – eliminate _"Not related to Shadow."_ from this error message.  It's particularly confusing here as the second sentence says _"Failed to parse shadow..."_.